subsys/net/ip/utils.c

```diff
@@ -766,3 +766,104 @@ int net_bytes_from_str(u8_t *buf, int buf_len, const char *src)
 
 	return 0;
 }
+
+int net_pkt_get_src_addr(struct net_pkt *pkt,
```

Comment:
It is a utility function. A very temporary one actually, because we should get rid of it asap as stated in the commit message. Only socket uses it because it's not taking advantage of net_context's recv_cb ip_hdr, proto_hdr parameters. I really don't want to see anybody else using that function, actually I might move it into socket directly

Reply:
Sure, having the prototype in net_private.h is ok, it is just that moving the implementation to utils.c seems a bit pointless here, we can keep it in net_pkt.c as the function name indicates.

Reply:
I was more radical: I move the function in socket.c and that's all. It's unused anywhere else, and removes code to maintain in utils unit test. And hopefully in some near future, socket will not need that function anymore as well.